packages/transport/src/create.ts

```diff
@@ -45,60 +54,108 @@ const makeAnyServiceImpl: CreateAnyServiceImpl = <S extends ServiceType>(
   return impl;
 };
 
-type ResolveReject<T> = Parameters<ConstructorParameters<typeof Promise<T>>[0]>;
-
 export const createChannelTransport = (s: ServiceType, port: MessagePort) => {
   const pending = new Map<
     ReturnType<typeof crypto.randomUUID>,
-    ResolveReject<JsonValue | ReadableStream<JsonValue>>
+    (response: TransportEvent) => void
   >();
 
+  let callbackError: Error | undefined;
+
+  let connected: boolean | undefined;
+
+  let connectionState: (a: boolean, r?: JsonValue) => void;
+  const providerAck = Promise.race([
+    new Promise<void>((_, reject) =>
+      setTimeout(
+        reject,
+        1000,
```

Comment:
Can we make this an constant at the top of the file (e.g. `const REQUEST_TIMEOUT = 1000`)? or perhaps a part of a config that gets passed in?